acousticbrainz/data/dump.py

```diff
@@ -122,54 +122,60 @@ def dump_db(location, threads=None, incremental=False, dump_id=None):
-        start_cond = "submitted > '%s'" % str(start_time) if start_time else ""
-        end_cond = "submitted <= '%s'" % str(end_time) if end_time else ""
-        if start_time and end_time:
-            where = "WHERE %s AND %s" % (start_cond, end_cond)
+    def generate_where(row_name, start_t=start_time, end_t=end_time):
+        """This function generates SQL WHERE clause that can be used to select
+        rows only within specified time frame using `row_name` as a reference.
+        """
+        if start_t or end_t:
+            start_cond = "%s > '%s'" % (row_name, str(start_t)) if start_t else ""
+            end_cond = "%s <= '%s'" % (row_name, str(end_t)) if end_t else ""
+            if start_t and end_t:
+                return "WHERE %s AND %s" % (start_cond, end_cond)
+            else:
+                return "WHERE %s%s" % (start_cond, end_cond)
         else:
-            where = "WHERE %s%s" % (start_cond, end_cond)
-    else:
-        where = ""
+            return ""
 
     # lowlevel
     with open(os.path.join(location, "lowlevel"), "w") as f:
         print(" - Copying table lowlevel...")
         cursor.copy_to(f, "(SELECT %s FROM lowlevel %s)" %
-                       (", ".join(_TABLES["lowlevel"]), where))
+                       (", ".join(_TABLES["lowlevel"]), generate_where("submitted")))
 
     # highlevel
     with open(os.path.join(location, "highlevel"), "w") as f:
         print(" - Copying table highlevel...")
         cursor.copy_to(f, "(SELECT %s FROM highlevel %s)" %
-                       (", ".join(_TABLES["highlevel"]), where))
+                       (", ".join(_TABLES["highlevel"]), generate_where("submitted")))
 
     # highlevel_json
     with open(os.path.join(location, "highlevel_json"), "w") as f:
         print(" - Copying table highlevel_json...")
-        query = """SELECT %s FROM highlevel_json WHERE id IN (
-                       SELECT data FROM highlevel %s
-                )""" % (", ".join(_TABLES["highlevel_json"]), where)
+        query = "SELECT %s FROM highlevel_json WHERE id IN (SELECT data FROM highlevel %s)" \
+                % (", ".join(_TABLES["highlevel_json"]), generate_where("submitted"))
         cursor.copy_to(f, "(%s)" % query)
 
-    # Copying tables that are always dumped with all rows
-    for table in _TABLES.keys():
-        if table in ("lowlevel", "highlevel", "highlevel_json"):
-            continue
-        print(" - Copying table %s..." % table)
-        with open(os.path.join(location, table), "w") as f:
-            cursor.copy_to(f, table, columns=_TABLES[table])
+    # statistics
+    with open(os.path.join(location, "statistics"), "w") as f:
+        print(" - Copying table statistics...")
+        cursor.copy_to(f, "(SELECT %s FROM statistics %s)" %
+                       (", ".join(_TABLES["statistics"]), generate_where("collected")))
+
+    # incremental_dumps
+    with open(os.path.join(location, "incremental_dumps"), "w") as f:
```

Comment:
does it make sense to dump the incremental dumps table?
 
- positive feature: anyone who has all of the dumps has a full dump of the db (except for our un-synched tables)
- negative feature: who actually cares about it?
- Maybe it's a good way of seeing what dump you're up to, so that you know which one to get next